lib/langchain/messages/chat_message.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Langchain
+  module Messages
+    # Generic messages that can be used in the `chat` method of the LLM adapter
+    # It is recommended that `assistants/messages/*_messages.rb` be unified with this one in the future.
+    class ChatMessage
+      attr_reader :role, :content
+
+      ROLES = [:system, :user, :assistant].freeze
```

Comment:
[These roles](https://github.com/patterns-ai-core/langchainrb/blob/main/lib/langchain/assistants/messages/google_gemini_message.rb#L7-L10) would be hidden, right? The user would only use these 3 when creating an instance, right? I think we'd still need one for `:tool`.

Reply:
I overlooked that because I hadn't thoroughly investigated LLMs other than OpenAI and Anthropic. In Gemini, it's model instead of assistant, right?
To maximize compatibility between LLMs, I think it's best not to allow the model role and have the LLM adapter layer convert the assistant role.

As mentioned earlier, I believe it's best to handle `tool` or `function` with specialized classes since the data they hold is different.
However, since this is an area where the differences in specifications between LLMs are significant, would it be okay to focus on supporting only user, system, and assistant roles in this PR for now?
I believe there will be no functional issues with proceeding incrementally, as the specification still allows receiving the previous Hash Array.